### PR TITLE
Exclude public option

### PR DESCRIPTION
**Pull request by vrujbr**
Hey, this PR adds support for a new flag `-exclude-public`.
It is a minimal thing. It doesn't support other access flags like internal, but it shows the way and gets the job I need done.
Btw the way you get `sourcekitd` stuff is quite unswifty, check out this code from Apple guys: https://github.com/apple/swift/blob/master/tools/SourceKit/tools/swift-lang/SourceKitdResponse.swift
You might just love it ;-)

### CODE REVIEW

**Comment by rockbruno** [comment]
Love the Swift bit! At the time I didn't really understood how SourceKit worked, so I got the API to work by ripping wrappers around the web... Definitely a good time to write our own version of the API!

I've seen you noticed that @hadiidbouk was already investigating this - I think that's a great way to do it but I do think it shares the problem where the methods declared inside public protocols will not be avoided, right?

File: swiftshield-Sources/AutomaticSwiftShield.swift [hunk 1]
```diff
@@ -108,6 +111,21 @@ extension AutomaticSwiftShield {
         guard let name = dict.getString(key: sourceKit.nameID)?.trueName, let usr = dict.getString(key: sourceKit.usrID) else {
             return nil
         }
+        
+        if excludePublic {
+            let attribtuesArray = dict.getDictionaryValue(key: sourceKit.attributesId)
```
Comment (rockbruno) [inline swiftshield-Sources/AutomaticSwiftShield.swift]: Nit: Typo on `attribtuesArray`

**Comment by vrujbr** [comment]
Yes I guess it should share the problem with protocols. Feel free to ignore this PR and wait for @hadiidbouk to finish his solution :-)

**Comment by rockbruno** [comment]
@hadiidbouk When are you going to push your changes? I think we can work together on this one 🙂

**Comment by rockbruno** [comment]
I'll close this because we have 3 PRs for the same feature, so I'm keeping only the original one.
By the way @vrujbr , I refactored the API to use Apple's wrapper and the results were great. Thanks again for showing me it! 🙂

**Comment by vrujbr** [comment]
Glad to help 👍

**Pull request closed without merging**